app.rb

```diff
@@ -0,0 +1,132 @@
+require './book'
+require './persons'
+require './rental'
+require './student'
+require './teacher'
+require './classroom'
+
+class App
+  def initialize
+    @book = []
+    @persons = []
+    @rentals = []
+  end
+
+  def open_console
+    puts 'Hello There, Welcome to our School library'
+    until list_of_options
+      input = gets.chomp
+      if input == '7'
+        puts 'Thank You for using my School Library!'
+        break
+      end
+
+      option input
+    end
+  end
+
+  def list_all_books
+    puts 'Database is empty! Add a book.' if @book.empty?
+    @book.each { |book| puts "[Book] Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def list_all_persons
+    puts 'Database is empty! Add a person.' if @persons.empty?
+    @persons.each { |person| puts "[#{person.class.name}] Name: #{person.name}, Age: #{person.age}, id: #{person.id}" }
+  end
+
+  def create_person
+    print 'To create a student, press 1, to create a teacher, press 2 : '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Invalid input. Try again'
+    end
+  end
+
+  def create_student
+    puts 'Create a new student'
+    print 'Enter student age: '
+    age = gets.chomp.to_i
+    print 'Enter name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+    case parent_permission
+    when 'n'
+      student = Student.new('undefined', age, name, parent_permission: false)
+      @persons << student
+      puts 'Student doesnt have parent permission, cant rent books'
+    when 'y'
+      student = Student.new('undefined', age, name, parent_permission: true)
```

Comment:
Kindly fix the issue with creating a student and a teacher as shown in the error message below:

![image](https://user-images.githubusercontent.com/20554583/216770260-837ad373-199c-41c1-ac0b-2c0202447d03.png)